python runner which use pulsar-client==3.2.0 cannot work with auth enabled pulsar cluster

error logs look like:

```
Error: l last): [2023-06-30 08:05:05 +0000] [ERROR] log.py: File "/pulsar/instances/python-instance/python_instance_main.py", line 239, in <module> [2023-06-30 08:05:05 +0000] [ERROR] log.py: main() [2023-06-30 08:05:05 +0000] [ERROR] log.py: File "/pulsar/instances/python-instance/python_instance_main.py", line 176, in main [2023-06-30 08:05:05 +0000] [ERROR] log.py: authentication = pulsar.Authentication(args.client_auth_plugin, args.client_auth_params) [2023-06-30 08:05:05 +0000] [ERROR] log.py: File "/pulsar/instances/python-instance/pulsar/__init__.py", line 278, in __init__ [2023-06-30 08:05:05 +0000] [ERROR] log.py: self.auth = _pulsar.Authentication(dynamicLibPath, authParamsString) [2023-06-30 08:05:05 +0000] [ERROR] log.py: TypeError [2023-06-30 08:05:05 +0000] [ERROR] log.py: : [2023-06-30 08:05:05 +0000] [ERROR] log.py: _pulsar.Authentication: No constructor defined!
Error: 6-30 08:05:05 +0000] [ERROR] log.py:   File "/pulsar/instances/python-instance/python_instance_main.py", line 239, in <module>
Error: 6-30 08:05:05 +0000] [ERROR] log.py: main()
Error: 6-30 08:05:05 +0000] [ERROR] log.py:   File "/pulsar/instances/python-instance/python_instance_main.py", line 176, in main
Error: 6-30 08:05:05 +0000] [ERROR] log.py: authentication = pulsar.Authentication(args.client_auth_plugin, args.client_auth_params)
Error: 6-30 08:05:05 +0000] [ERROR] log.py:   File "/pulsar/instances/python-instance/pulsar/__init__.py", line 278, in __init__
Error: 6-30 08:05:05 +0000] [ERROR] log.py: self.auth = _pulsar.Authentication(dynamicLibPath, authParamsString)
Error: 6-30 08:05:05 +0000] [ERROR] log.py: TypeError
Error: 6-30 08:05:05 +0000] [ERROR] log.py: :
Error: 6-30 08:05:05 +0000] [ERROR] log.py: _pulsar.Authentication: No constructor defined!'
Error: 023-06-30 08:05:05 +0000] [ERROR] log.py: Traceback (most recent call last):
Error: 6-30 08:05:05 +0000] [ERROR] log.py:   File "/pulsar/instances/python-instance/python_instance_main.py", line 239, in <module>
Error: 6-30 08:05:05 +0000] [ERROR] log.py: main()
Error: 6-30 08:05:05 +0000] [ERROR] log.py:   File "/pulsar/instances/python-instance/python_instance_main.py", line 176, in main
Error: 6-30 08:05:05 +0000] [ERROR] log.py: authentication = pulsar.Authentication(args.client_auth_plugin, args.client_auth_params)
Error: 6-30 08:05:05 +0000] [ERROR] log.py:   File "/pulsar/instances/python-instance/pulsar/__init__.py", line 278, in __init__
Error: 6-30 08:05:05 +0000] [ERROR] log.py: self.auth = _pulsar.Authentication(dynamicLibPath, authParamsString)
Error: 6-30 08:05:05 +0000] [ERROR] log.py: TypeError
Error: 6-30 08:05:05 +0000] [ERROR] log.py: :
```

--- a/.ci/clusters/values_runner_images.yaml
+++ b/.ci/clusters/values_runner_images.yaml
@@ -172,9 +172,9 @@ functions:
     disk: 1048576000
   functionRuntimeFactoryConfigs:
     functionDockerImages:
-      JAVA: "streamnative/pulsar-functions-java-runner:2.9.2.23"
-      PYTHON: "streamnative/pulsar-functions-python-runner:2.9.2.23"
-      GO: "streamnative/pulsar-functions-go-runner:2.9.2.23"
+      JAVA: "localhost:5000/pulsar-functions-java-runner:2.9.2.23"
+      PYTHON: "localhost:5000/pulsar-functions-python-runner:2.9.2.23"
+      GO: "localhost:5000/pulsar-functions-go-runner:2.9.2.23"
     functionInstanceMinResources:
       cpu: 0.1
       ram: 10485760

--- a/.github/workflows/test-function-runner.yml
+++ b/.github/workflows/test-function-runner.yml
@@ -41,8 +41,10 @@ jobs:
             .ci/cleanup.sh
 
       - name: Build runner images
+        env:
+          DOCKER_REPO: localhost:5000
         run: |
-          PULSAR_IMAGE_TAG=2.9.2.23 PULSAR_IMAGE=streamnative/pulsar-all KIND_PUSH=true images/build.sh
+          PULSAR_IMAGE_TAG=2.9.2.23 PULSAR_IMAGE=streamnative/pulsar-all KIND_PUSH=false images/build.sh
 
       - name: Verify function runner
         run: |

--- a/images/build.sh
+++ b/images/build.sh
@@ -72,6 +72,15 @@ if [ "$KIND_PUSH" = true ] ; then
     kind load docker-image "${DOCKER_REPO}"/${PULSARCTL_GO_RUNNER}:"${RUNNER_TAG}" --name $cluster
   done
 fi
+
+if [ "$DOCKER_REPO" = "localhost:5000" ]; then
+  docker push "${DOCKER_REPO}"/${JAVA_RUNNER}:"${RUNNER_TAG}"
+  docker push "${DOCKER_REPO}"/${PULSARCTL_JAVA_RUNNER}:"${RUNNER_TAG}"
+  docker push "${DOCKER_REPO}"/${PYTHON_RUNNER}:"${RUNNER_TAG}"
+  docker push "${DOCKER_REPO}"/${PULSARCTL_PYTHON_RUNNER}:"${RUNNER_TAG}"
+  docker push "${DOCKER_REPO}"/${GO_RUNNER}:"${RUNNER_TAG}"
+  docker push "${DOCKER_REPO}"/${PULSARCTL_GO_RUNNER}:"${RUNNER_TAG}"
+fi
 #
 #if [ "$CI_TEST" = true ] ; then
 #  echo "apply images to function mesh ci yaml"

--- a/images/pulsar-functions-python-runner/Dockerfile
+++ b/images/pulsar-functions-python-runner/Dockerfile
@@ -41,6 +41,10 @@ WORKDIR /pulsar
 RUN if [ -f "/pulsar/bin/install-pulsar-client-37.sh" ]; then /pulsar/bin/install-pulsar-client-37.sh || true ; fi
 RUN if [ -f "/pulsar/bin/install-pulsar-client.sh" ]; then /pulsar/bin/install-pulsar-client.sh || pip3 install 'pulsar-client[all]==3.1.0' ; fi
 
+# this dir is duplicate with the installed pulsar-client pip package, and maybe not compatible with the `_pulsar`(ths .so library package)
+RUN rm -rf /pulsar/instances/python-instance/pulsar/ \
+     && sed -i "s/serde.IdentitySerDe/pulsar.functions.serde.IdentitySerDe/g" /pulsar/instances/python-instance/python_instance.py
+
 
 USER $USER
 # a temp solution from https://github.com/apache/pulsar/pull/15846 to fix python protobuf version error

--- a/images/pulsar-functions-python-runner/Dockerfile
+++ b/images/pulsar-functions-python-runner/Dockerfile
@@ -41,9 +41,10 @@ WORKDIR /pulsar
 RUN if [ -f "/pulsar/bin/install-pulsar-client-37.sh" ]; then /pulsar/bin/install-pulsar-client-37.sh || true ; fi
 RUN if [ -f "/pulsar/bin/install-pulsar-client.sh" ]; then /pulsar/bin/install-pulsar-client.sh || pip3 install 'pulsar-client[all]==3.1.0' ; fi
 
-# this dir is duplicate with the installed pulsar-client pip package, and maybe not compatible with the `_pulsar`(ths .so library package)
+# this dir is duplicate with the installed pulsar-client pip package, and maybe not compatible with the `_pulsar`(the .so library package)
 RUN rm -rf /pulsar/instances/python-instance/pulsar/ \
-     && sed -i "s/serde.IdentitySerDe/pulsar.functions.serde.IdentitySerDe/g" /pulsar/instances/python-instance/python_instance.py
+     && sed -i "s/serde.IdentitySerDe/pulsar.functions.serde.IdentitySerDe/g" /pulsar/instances/python-instance/python_instance.py \
+     && sed -i "s/serde.IdentitySerDe/pulsar.functions.serde.IdentitySerDe/g" /pulsar/instances/python-instance/contextimpl.py
 
 
 USER $USER

--- a/images/pulsar-functions-python-runner/pulsarctl.Dockerfile
+++ b/images/pulsar-functions-python-runner/pulsarctl.Dockerfile
@@ -34,6 +34,10 @@ RUN if [ -d "/pulsar/cpp-client" ]; then apt-get update \
 RUN if [ -f "/pulsar/bin/install-pulsar-client-37.sh" ]; then /pulsar/bin/install-pulsar-client-37.sh || pip3 install 'pulsar-client[all]==3.1.0' ; fi
 RUN if [ -f "/pulsar/bin/install-pulsar-client.sh" ]; then /pulsar/bin/install-pulsar-client.sh || pip3 install 'pulsar-client[all]==3.1.0' ; fi
 
+# this dir is duplicate with the installed pulsar-client pip package, and maybe not compatible with the `_pulsar`(ths .so library package)
+RUN rm -rf /pulsar/instances/python-instance/pulsar/ \
+     && sed -i "s/serde.IdentitySerDe/pulsar.functions.serde.IdentitySerDe/g" /pulsar/instances/python-instance/python_instance.py
+
 WORKDIR /pulsar
 
 USER $USER

--- a/images/pulsar-functions-python-runner/pulsarctl.Dockerfile
+++ b/images/pulsar-functions-python-runner/pulsarctl.Dockerfile
@@ -34,9 +34,10 @@ RUN if [ -d "/pulsar/cpp-client" ]; then apt-get update \
 RUN if [ -f "/pulsar/bin/install-pulsar-client-37.sh" ]; then /pulsar/bin/install-pulsar-client-37.sh || pip3 install 'pulsar-client[all]==3.1.0' ; fi
 RUN if [ -f "/pulsar/bin/install-pulsar-client.sh" ]; then /pulsar/bin/install-pulsar-client.sh || pip3 install 'pulsar-client[all]==3.1.0' ; fi
 
-# this dir is duplicate with the installed pulsar-client pip package, and maybe not compatible with the `_pulsar`(ths .so library package)
+# this dir is duplicate with the installed pulsar-client pip package, and maybe not compatible with the `_pulsar`(the .so library package)
 RUN rm -rf /pulsar/instances/python-instance/pulsar/ \
-     && sed -i "s/serde.IdentitySerDe/pulsar.functions.serde.IdentitySerDe/g" /pulsar/instances/python-instance/python_instance.py
+     && sed -i "s/serde.IdentitySerDe/pulsar.functions.serde.IdentitySerDe/g" /pulsar/instances/python-instance/python_instance.py \
+     && sed -i "s/serde.IdentitySerDe/pulsar.functions.serde.IdentitySerDe/g" /pulsar/instances/python-instance/contextimpl.py
 
 WORKDIR /pulsar
 

